src/main/java/com/artipie/api/BaseRest.java

```diff
@@ -41,6 +44,17 @@ protected Handler<RoutingContext> errorHandler(final int code) {
         };
     }
 
+    /**
+     * Read body as JsonObject.
+     * @param context RoutingContext
+     * @return JsonObject
+     */
+    protected static JsonObject readJsonObject(final RoutingContext context) {
+        return (JsonObject) (Json.createReader(
+            new StringReader(context.body().asString())
+        ).read());
+    }
+
```

Comment:
sorry, I've missed this static `BaseRest#validator(com.artipie.api.Validator...)` method which can be easily remade into normal object - decorator for `Validator`. The class can be a subclass of `Validator` and called `All`. Here is an [example](https://github.com/artipie/http/blob/45cc8611213da923e2e35ada3e8dfb676214e5ab/src/main/java/com/artipie/http/rt/RtRule.java#L68) from http, please, follow it.